src/Contract/Utxos.purs

```diff
@@ -2,31 +2,43 @@
 -- | input to transaction output. Furthermore, a helper to get the utxos at
 -- | a given `Address` is defined.
 module Contract.Utxos
-  ( utxosAt
+  ( getUtxo
   , getWalletBalance
   , module Transaction
+  , utxosAt
   ) where
 
 import Prelude
 
-import Contract.Monad (Contract, wrapContract, liftContractM)
+import Contract.Monad (Contract, liftContractM, wrapContract)
+import Contract.Transaction (TransactionInput, TransactionOutput)
 import Control.Monad.Reader.Class (asks)
 import Data.Maybe (Maybe(Just, Nothing), maybe)
 import Data.Newtype (unwrap)
-import Plutus.Conversion (fromPlutusAddress, toPlutusUtxoM)
+import Plutus.Conversion (fromPlutusAddress, toPlutusTxOutput, toPlutusUtxoM)
 import Plutus.Conversion.Value (toPlutusValue)
 import Plutus.Types.Address (Address)
 import Plutus.Types.Transaction (UtxoM(UtxoM)) as Transaction
 import Plutus.Types.Value (Value)
-import QueryM.Utxos (getWalletBalance, utxosAt) as Utxos
+import Prim.TypeError (class Warn, Text)
+import QueryM.Utxos (getUtxo, getWalletBalance, utxosAt) as Utxos
 
 -- | This module defines query functionality via Ogmios to get utxos.
 
 -- | Gets utxos at an (internal) `Address` in terms of a Plutus `Address`.
 -- | Results may vary depending on `Wallet` type. See `QueryM` for more details
 -- | on wallet variance.
+-- |
+-- | NOTE: Querying for UTxOs by address is deprecated. See
+-- | [here](https://github.com/CardanoSolutions/ogmios/issues/199).
 utxosAt
-  :: forall (r :: Row Type). Address -> Contract r (Maybe Transaction.UtxoM)
+  :: forall (r :: Row Type)
+   . Warn
+       ( Text
+           "`utxosAt`: Querying for UTxOs by address is deprecated. See https://github.com/CardanoSolutions/ogmios/issues/199."
```

Comment:
Please add this link instead: https://github.com/Plutonomicon/cardano-transaction-lib/issues/536

I think it's better to point users to our issue, to make it clear how we are going to handle that problem in the future.